src/eigen/diag_lobpcg_hyper.jl

```diff
@@ -9,8 +9,10 @@ function lobpcg_hyper(A, X0; maxiter=100, prec=nothing,
     result = LOBPCG(A, X0, I, prec, tol, maxiter; n_conv_check=n_conv_check, kwargs...)
 
     n_conv_check === nothing && (n_conv_check = size(X0, 2))
+
     converged = maximum(result.residual_norms[1:n_conv_check]) < tol
     iterations = size(result.residual_history, 2) - 1
+    λ = Array(result.λ) #TODO: offload this to gpu? Careful then, as self_consistent_field's eigenvalues will be a CuArray -> due to the Smearing.occupation function, occupation will also be a CuArray, so no scalar indexing (in ene_ops, in compute_density...)
```

Comment:
Can't we just add this at the end of `diagonalize_all_kblocks`? High level objects go in and a tuple goes out with all necessary fields: we simply transfer to CPU when all blocks have been diagonalized.
Or we could say we want `λ` to be on the GPU but I'm not sure it's necessary, and it will require quite some work as this means that `occupation` will be on the GPU as well (so no scalar indexing, which is something done a lot for this array).